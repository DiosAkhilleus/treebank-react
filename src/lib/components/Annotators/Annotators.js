import React from 'react';
import { object } from 'prop-types';

import styles from './Annotators.module.scss';

const annotatorFields = [
  'short',
  'name',
  'address',
  'uri',
];

const annotatorsFromJson = (json) => {
  const annotators = json.treebank.annotator;

  if (annotators) {
    return annotators.map((annotator) => annotatorFields
      .map((field) => ({ title: field, value: annotator[field] ? annotator[field][0] : '' }))
      .filter(({ value }) => value !== ''));
  }

  return [];
};

// eslint-disable-next-line react/prop-types
const renderField = ({ title, value }, ii, jj) => (
  <div key={`annotator-field-${ii}-${jj}`} className={styles.container}>
    <dt className={styles.dt}>{title}</dt>
    <dd className={styles.dd}>{value}</dd>
  </div>
);

const Annotators = ({ json }) => (
  annotatorsFromJson(json).map((annotator, ii) => (
    // eslint-disable-next-line react/no-array-index-key
    <div key={`annotator-${ii}`} className={styles.annotators}>
      <dl className={styles.dl}>
        {annotator.map((field, jj) => renderField(field, ii, jj))}
      </dl>
    </div>
  ))
);

Annotators.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  json: object.isRequired,
};

export default Annotators;
