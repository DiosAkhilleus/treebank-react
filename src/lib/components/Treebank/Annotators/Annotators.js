import React from 'react';

import TreebankContext from '../treebank-context';

import Annotators from '../../Annotators';

const AnnotatorsWithContext = () => (
  <TreebankContext.Consumer>
    {({ json }) => <Annotators json={json} />}
  </TreebankContext.Consumer>
);

export default AnnotatorsWithContext;
