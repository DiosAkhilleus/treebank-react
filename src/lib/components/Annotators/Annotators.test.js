import React from 'react';
import renderer from 'react-test-renderer';

import Annotators from './Annotators';

it('renders the annotators', () => {
  const json = {
    treebank: {
      annotator: [
        {
          short: 'T', name: 'Test User', address: 'test@example.com', uri: 'http://example.com',
        },
        { uri: 'http://example.com/only-uri' },
      ],
    },
  };

  const component = (
    <Annotators json={json} />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
