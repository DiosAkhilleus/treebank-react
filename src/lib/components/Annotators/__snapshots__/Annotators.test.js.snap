// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the annotators 1`] = `
Array [
  <div
    className="annotators"
  >
    <dl
      className="dl"
    >
      <div
        className="container"
      >
        <dt
          className="dt"
        >
          short
        </dt>
        <dd
          className="dd"
        >
          T
        </dd>
      </div>
      <div
        className="container"
      >
        <dt
          className="dt"
        >
          name
        </dt>
        <dd
          className="dd"
        >
          T
        </dd>
      </div>
      <div
        className="container"
      >
        <dt
          className="dt"
        >
          address
        </dt>
        <dd
          className="dd"
        >
          t
        </dd>
      </div>
      <div
        className="container"
      >
        <dt
          className="dt"
        >
          uri
        </dt>
        <dd
          className="dd"
        >
          h
        </dd>
      </div>
    </dl>
  </div>,
  <div
    className="annotators"
  >
    <dl
      className="dl"
    >
      <div
        className="container"
      >
        <dt
          className="dt"
        >
          uri
        </dt>
        <dd
          className="dd"
        >
          h
        </dd>
      </div>
    </dl>
  </div>,
]
`;
